smarts/core/mission_planner.py

```diff
@@ -193,6 +200,58 @@ def _edge_ids(self, pose: Pose, lane_id: str = None):
 
         return edge_ids
 
+    def cut_in_waypoints(self, pose: Pose, sim, vehicle):
+        radius = 30
+        neighborhood_vehicles = sim.neighborhood_vehicles_around_vehicle(
+            vehicle=vehicle, radius=radius
+        )
+
+        if not neighborhood_vehicles:
+            return []
+
+        nei_vehicle = neighborhood_vehicles[0]
+        position = pose.position[:2]
+        lane = self._road_network.nearest_lane(position)
+        target_position = nei_vehicle.pose.position[:2]
+        target_lane = self._road_network.nearest_lane(target_position)
+
+        offset = self._road_network.offset_into_lane(lane, position)
+        target_offset = self._road_network.offset_into_lane(
```

Comment:
I seem to remember we may have to do some improvements on the `offset_into_lane()` in the future since it currently relies on the sumolib implementation... which is not fully truthful when it comes to curvy lanes.